controllers/idler/idler_controller.go

```diff
@@ -299,6 +300,24 @@ func (r *Reconciler) scaleDeploymentToZero(logger logr.Logger, namespace string,
 					return true, nil
 				}
 
+				return false, err
+			} else if errors.IsInternalError(err) { // Failed to get existing scale resource - try to set a new one
```

Comment:
Is the internal error the only way to detect a missing optional replica field?
If so the let's at least log an infra message and include the original error message there too.

Reply:
yes, unfortunately this is being reported as a 500 internal error `Internal error occurred: the spec replicas field \".spec.replicas\" does not exist`

I am afraid that this will be the default path for almost all resources with a scale sub resource as a missing replicas field is equal to the default replicas=1. Do we really want all of those errors to be logged then?

Reply:
Ok. Let's keep it as is. But please add a test :)

Reply:
I have added a unit test and I narrowed down the internal server error to have the specific error message saying that the spec.replicas field does not exist.